### PR TITLE
Attempt to fix 3p-ad-impl flaky test

### PR DESCRIPTION
**Pull request by zhouyx**
Attempt to fix the flaky test. The original test didn't handle the timer stub correctly somehow.

### CODE REVIEW

File: extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js [hunk 1]
```diff
@@ -564,7 +571,7 @@ describes.realWin('amp-ad-3p-impl', {
           // Right margin is 9px from containerContainer and 25px from
           // container.
           expect(element.style.marginLeft).to.be.equal('');
-          expect(element.style.marginRight).to.be.equal('-49px');
+          expect(element.style.marginRight).to.be.equal('-34px');
```
Comment (lannka) [inline extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js]: was the testing continuously failing?

File: extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js [hunk 2]
```diff
@@ -317,8 +321,11 @@ describes.realWin('amp-ad-3p-impl', {
     });
 
     it('should only allow rendering one ad per second', function* () {
-      const clock = lolex.install({target: win});
+      const clock = lolex.install({
+        target: win, toFake: ['Date', 'setTimeout', 'clearTimeout']});
+      installTimerService(win);
```
Comment (lannka) [inline extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js]: is this needed?

File: extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js (hunk 1 above)
Comment (zhouyx) [reply extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js]: hmmmm. According to the comment we should put -34px. 
The test is very closely related to the browser size. For some reason, it keeps failing when I was debugging yesterday. But somehow it works today....... Anyhow I reverted the change.

File: extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js (hunk 2 above)
Comment (zhouyx) [reply extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js]: nope. removed.

**Pull request merged**